homeassistant/components/tuya/diagnostics.py

```diff
@@ -84,8 +84,8 @@ def _async_device_as_dict(hass: HomeAssistant, device: TuyaDevice) -> dict[str,
 
     # Base device information, without sensitive information.
     data = {
-        "name": device.model,
-        "model": device.model,
+        "name": device.name,
+        "model": f"{device.product_name} ({device.product_id})",
```

Comment:
I think it would be better to fall back to `None` if the property doesn't exist.
Adding the current suggestion has no use, as those values are already passed.

Reply:
Ok.